tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/DirectSubmission/LoggerFactoryIntegrationCommon.cs

```diff
@@ -81,13 +81,22 @@ internal static void AddDirectSubmissionLoggerProvider(TLoggerFactory loggerFact
         }
 
         // Internal for testing
-        internal static object AddDirectSubmissionLoggerProvider(TLoggerFactory loggerFactory, DirectSubmissionLoggerProvider provider)
+        internal static object? AddDirectSubmissionLoggerProvider(TLoggerFactory loggerFactory, DirectSubmissionLoggerProvider provider)
         {
+            if (ProviderInterfaces is null)
+            {
+                // there was a problem loading the assembly for some reason
+                return null;
+            }
+
             var proxy = provider.DuckImplement(ProviderInterfaces);
-            var loggerFactoryProxy = loggerFactory.DuckCast<ILoggerFactory>();
-            loggerFactoryProxy.AddProvider(proxy);
+            if (loggerFactory is not null)
+            {
+                var loggerFactoryProxy = loggerFactory.DuckCast<ILoggerFactory>();
+                loggerFactoryProxy.AddProvider(proxy);
```

Comment:
Should we be checking that `loggerFactoryProxy != null` before calling `AddProvider`?

Reply:
I think if we are in a `DuckCast` scenario, if `loggerFactory` is not null then `loggerFactoryProxy` shouldn't be null.